conmon-rs/common/proto/conmon.capnp

```diff
@@ -169,6 +171,18 @@ interface Conmon {
 
     createNamespaces @6 (request: CreateNamespacesRequest) -> (response: CreateNamespacesResponse);
 
+    ###############################################
+    # StartFdSocket
+    struct StartFdSocketRequest {
+        metadata @0 :Metadata; # Standard metadata to carry.
+    }
+
```

Comment:
should we just always start it when conmon-rs starts? that will prevent container creation requests from needing to check it's created before sending some down

Reply:
Some clients (e.g. CRI-O) do not need file descriptor passing at all or only for some containers (e.g. podman), therefore they also don't need the extra socket.

And when automatic shutdown is implemented, we can also stop the socket after a few seconds of inactivity.

Reply:
yeah I just think not starting it on startup makes some weird situations. does podman need to call StartFdSocket each time it runs to make sure? should the client keep track of it? how would the client restore that information? If we always start, we eliminate a lot of edge cases, for the cost of one unix socket running for a client that doesn't need it

Reply:
The current implementation stores no state in the client and always fetches the path from the server (and thereby starts the socket). This introduces an extra RPC call, but there should be no edge cases.

Reply:
I see the benefits of both solutions and am not opinionated. :thinking:

Reply:
let's leave as-is then